src/base/rule_no_invalid_default_values.re

```diff
@@ -0,0 +1,90 @@
+module Visitor: Traversal_utils.VisitorSig = {
+  open Traversal_utils;
+  open Schema;
+  open Source_pos;
+
+  include AbstractVisitor;
+
+  type t = {
+    mutable argType: option(string),
+    mutable nameSpan: option(span),
+  };
+  let make_self = () => {argType: None, nameSpan: None};
+
+  let enter_argument =
+      (
+        self,
+        ctx,
+        (name: spanning(string), arg_type: option(type_ref), value),
+      ) => {
+    switch (
+      arg_type
+      |> Option.map(innermost_name)
+      |> Option.flat_map(lookup_type(ctx.schema))
+    ) {
+    | Some(type_meta) when type_meta |> is_type_default =>
+      self.argType = Some(type_meta |> type_name);
+      self.nameSpan = Some(name.span);
+    | Some(_)
+    | None => ()
+    };
+  };
+
+  let report_error = (ctx, expected, received, span) => {
+    Context.push_error(
+      ctx,
+      span,
+      Printf.sprintf(
+        "Invalid default value. Expected \"%s\" but received \"%s\".",
+        expected,
+        received,
+      ),
+    );
+  };
+
+  let check_input = (ctx, expected, received, span) => {
+    switch (expected) {
+    | None => ()
+    | Some(expected) when expected == received => ()
+    | Some(expected) => report_error(ctx, expected, received, span)
+    };
+  };
+
+  let enter_int_value = (self, ctx, value) => {
```

Comment:
Can make this shorter by having `received` as the last argument of `check_input` and currying

Reply:
I'm not too sure about it. I just tried a few variations but the special cases always catch up and it ends up being more difficult to reason about with almost no gains for verbosity.